app/models/agreement_letter.rb

```diff
@@ -65,49 +65,48 @@ def set_path
   end
 
   private
-    def valid_file?(file)
-      if !is_file?(file)
-        errors.add(I18n.t('agreement_letters.file_error'), I18n.t("agreement_letters.not_a_file"))
-        false
-      elsif too_big?(file)
-        errors.add(I18n.t('agreement_letters.file_error'), I18n.t("agreement_letters.file_too_big"))
-        false
-      elsif wrong_filetype?(file)
-        errors.add(I18n.t('agreement_letters.file_error'), I18n.t("agreement_letters.wrong_filetype"))
-        false
-      elsif unable_to_open?(file)
-        errors.add(I18n.t('agreement_letters.file_error'), I18n.t("agreement_letters.corrupt_document"))
-        false
-      else
-        true
-      end
-    end
-    
-    def is_file?(file)
-      file.respond_to?(:open) && file.respond_to?(:content_type) && file.respond_to?(:size)
-    end
-
-    def too_big?(file)
-      file.size > MAX_SIZE
-    end
-
-    def wrong_filetype?(file)
-      file.content_type != ALLOWED_MIMETYPE
-    end
 
-  def unable_to_open? file
-    begin
-      PDF::Inspector::Page.analyze_file(file.open)
+  def valid_file?(file)
+    if !is_file?(file)
+      errors.add(I18n.t('agreement_letters.file_error'), I18n.t('agreement_letters.not_a_file'))
+      false
+    elsif too_big?(file)
+      errors.add(I18n.t('agreement_letters.file_error'), I18n.t('agreement_letters.file_too_big'))
+      false
+    elsif wrong_filetype?(file)
+      errors.add(I18n.t('agreement_letters.file_error'), I18n.t('agreement_letters.wrong_filetype'))
+      false
+    elsif unable_to_open?(file)
+      errors.add(I18n.t('agreement_letters.file_error'), I18n.t('agreement_letters.corrupt_document'))
       false
-    rescue PDF::Reader::UnsupportedFeatureError, PDF::Reader::MalformedPDFError
```

Comment:
The presentation of this diff is a little bit weired, but I think this line is now [here](https://github.com/hpi-schuelerklub/workshop-portal/pull/676/files/02458dece54bc25584346eccc9e96c1895f77d93#diff-b886d6b9c92f8c631ec0fad6268523ebR102)